packets.js

```diff
@@ -127,21 +127,57 @@ runknown.encodingLength = function (data) {
 
 var rtxt = {}
 
-rtxt.encode = function (s, buf, offset) {
-  str.encode(s, buf, offset + 2)
-  buf.writeUInt16BE(str.encode.bytes, offset)
-  rtxt.encode.bytes = str.encode.bytes + 2
+rtxt.encode = function (data, buf, offset) {
+  var oldOffset = offset
+  offset += 2
+
+  if (!data) data = {}
+  var keys = Object.keys(data)
+
+  if (keys.length === 0) {
+    str.encode('', buf, offset)
+    offset += str.encode.bytes
+  }
+
+  keys.forEach(function (key) {
+    var s = key + '=' + data[key]
```

Comment:
I just re-read the RFC and it actually says:

> If there is no '=' in a DNS-SD TXT record string, then it is a boolean attribute, simply identified as being present, with no value.

So in case there is an equal sign, I guess it should actually be interpreted as an empty string. 

But your point is still important. I guess you want the encoder to be an exact mirror of the decoder, so that the same raw TXT record can be decoded and then encoded and produce the same TXT record.

I'll update the PR to follow those rules :smiley: